src/compute/generic.ts

```diff
@@ -7,14 +7,29 @@ import { Filters, generateFiltersQuery } from '../filters'
 import { ratioToPercentage } from './common'
 import { getEntity } from '../helpers'
 import { getParticipationByYearMap } from './demographics'
+import { useCache } from '../caching'
 
-interface TermAggregationByYearOptions {
+const firstYear = 2016
+const currentYear = new Date().getFullYear()
+const allYears = _.range(firstYear, currentYear)
+
+interface TermAggregationAllYearsOptions {
+    // filter aggregations
+    filters?: Filters
+    sort?: string
+    order?: -1 | 1
+    cutoff?: number
+    limit?: number
+}
+
+interface TermAggregationSingleYearOptions {
```

Comment:
You could extend `TermAggregationAllYearsOptions` and just add the `year` property, might be easier for maintenance and consistency.